### ISSUE TITLE
Genericize Crafting Table Detection

### ISSUE THREAD

**Issue by hammy275** [hammy3502/immersive-mc#352]
Already done for other blocks, should do it for crafting tables, too.

**Comment by hammy275** [comment]
Done and done!

**Comment by hammy275** [comment]
Had to undo the work from this due to a bug with smithing tables (see #371 ). I'm thinking of using a user-customizable whitelist in the future, but until then, this issue is back open.

**Comment by hammy275** [comment]
I should use tags for this. Great system already in the game. I can internally keep a list of all crafting tables I know are supported, and it should let authors add their own through the vanilla tag system (will have to write a wiki article for that).

**Comment by hammy275** [comment]
I should be able to detect crafting tables using each modloader's pre-existing crafting table tag combined with the new convention one (https://github.com/FabricMC/fabric/blob/e521378444edf1a1f4d41311b8be7edf2c17aef1/fabric-convention-tags-v2/src/generated/resources/data/c/tags/block/player_workstations/crafting_tables.json).

**Comment by EyaeM** [comment]
most crafting tables are not supported only (in my testing) the vanilla one works

**Comment by hammy275** [comment]
Yes, this is planned to be fixed. At the moment, ImmersiveMC directly checks if the block is the vanilla crafting table, as other forms of detection that don't use tags end up catching other blocks that are built similarly (see the smithing table issue mentioned above).

**Comment by hammy275** [comment]
Tags are admittedly kind of a mess, and I worry about picking up blocks marked as crafting tables that wouldn't actually want to be used with this Immersive.

Instead, I think I should pull a page from Visual Workbench, and go with everything that's a crafting table block but also isn't a fletching table block or a smithing table block.
